map.go

```diff
@@ -892,12 +894,21 @@ func (m *Map) Freeze() error {
 	return nil
 }
 
-func (m *Map) populate(contents []MapKV) error {
-	for _, kv := range contents {
+// populate populates the Map according to the Contents specified
+// in spec and freezes the Map if requested by spec.
+func (m *Map) populate(spec *MapSpec) error {
```

Comment:
Maybe `finalize`? `populate` makes me forget that the map might be frozen after this.